### ISSUE TITLE
[BUG] Inconsitent index if TYPO3 Site base variants are used

### ISSUE THREAD

**Issue by sascha-egerer** [TYPO3-Solr/ext-solr#2578]
**Describe the bug**
The index is currently based on the sitehash which is build by using the TYPO3 site "base". 
TYPO3 has the possibility to define "baseVariants" which means the domain does change if some condition is met. I do use this to have a different domain for the backend than for the frontend. But this ends up in having documents in the index with 2 different sitehash values as the document is index based the "base" for the current context. This does also mean that you get different results for the same site if using different domains.

**To Reproduce**
1. Create a TYPO3 site with a baseVariant that is used if some condition does match
2. Initialize the index queue
3. Run the indexing from the CLI
4. Open TYPO3 with the domain used in the baseVariant and trigger the indexing via the indexing module

**Expected behavior**
There should be one document in solr with a unique sitehash as there is only one "site".
The siteHash should be build based on the "site identifier" and not based on the "base".

**Used versions (please complete the following information):**
 - TYPO3 Version: 9.5.15
 - EXT:solr Version: master

**Additional context**

**Comment by avogt1701** [comment]
Thats correct for a baseVariants use case with different contexts (dev/prod). In our case we have multiple base variants for the same context (Production). 

I think it would be better to set the solr field "site" to the site identifier and add the application context as additional site hash field.

**Comment by dmitryd** [comment]
Hopefully it will not slow done queries.
